packages/data-broker/src/DataBroker/DataBroker.ts

```diff
@@ -323,24 +301,15 @@ export class DataBroker {
     });
   }
 
-  public async pull(
-    dataSourceSpec: DataSourceSpec<'dataElement'>,
-    options: Record<string, unknown>,
-  ): Promise<RawAnalyticResults>;
-  public async pull(
-    dataSourceSpec: DataSourceSpec<'dataGroup'>,
-    options: Record<string, unknown>,
-  ): Promise<EventResults>;
-  public async pull(
-    dataSourceSpec: DataSourceSpec<'syncGroup'>,
-    options: Record<string, unknown>,
-  ): Promise<SyncGroupResults>;
-  public async pull(dataSourceSpec: DataSourceSpec, options: Record<string, unknown> = {}) {
-    const dataSources = await this.fetchDataSources(dataSourceSpec);
-    const { type } = dataSourceSpec;
+  public async pullAnalytics(
+    dataElementCodes: string[],
+    options: PullOptions,
+  ): Promise<AnalyticResults> {
+    const dataElements = await fetchDataElements(this.models, dataElementCodes);
     const validatedOptions = setOrganisationUnitCodes(options);
 
-    const pulls = await this.getPulls(dataSources, validatedOptions.organisationUnitCodes);
+    const pulls = await this.getPulls(dataElements, validatedOptions.organisationUnitCodes);
+    const type = this.getDataSourceTypes().DATA_ELEMENT;
     const nestedResults = await Promise.all(
       pulls.map(({ dataSources: dataSourcesForThisPull, serviceType, dataServiceMapping }) => {
         return this.pullForServiceAndType(
```

Comment:
`pullForServiceAndType` should also be split - inline the data source specific logic, and perhaps used a shared function for shared parts of the logic.

I'll leave this for a future PR to keep the PR simple